packages/core/src/services/providerLogs/addMessages.ts

```diff
@@ -62,13 +68,11 @@ export async function addMessages({
     )
   }
 
-  let responseResolve: (value?: ChainStepResponse<StreamType>) => void
+  let responseResolve: (value: ChainResponse<StreamType>) => void
```

Comment:
hmm you changed the response type of this service yet no test is failing, maybe a good opportunity to add tests for this service.